lib/plugins/filter/after_render/injector.js

```diff
@@ -1,53 +1,38 @@
 'use strict';
 
+const { Cache } = require('hexo-util');
+const cache = new Cache();
+
 function injectFilter(data, locals = { page: {} }) {
   const { injector: _Injector } = this.extend;
 
-  const is = input => {
-    let result = true;
+  const current = () => {
     const { page } = locals;
 
-    switch (input) {
-      case 'home':
-        result = Boolean(page.__index);
-        break;
-
-      case 'post':
-        result = Boolean(page.__post);
-        break;
-
-      case 'page':
-        result = Boolean(page.__page);
-        break;
-
-      case 'archive':
-        result = Boolean(page.archive);
-        break;
-
-      case 'category':
-        result = Boolean(page.category);
-        break;
-
-      case 'tag':
-        result = Boolean(page.tag);
-        break;
-
-      default:
-        result = true;
-    }
-
-    return result;
+    if (page.__index) return 'home';
+    if (page.__post) return 'post';
+    if (page.__page) return 'page';
+    if (page.archive) return 'archive';
+    if (page.category) return 'category';
+    if (page.tag) return 'tag';
+    return 'default';
   };
 
   function injector(data, pattern, flag, isBegin = true) {
     if (data.includes(`hexo injector ${flag}`)) return data;
 
-    const arr = _Injector.get(flag).filter(i => is(i.to)).map(i => i.value);
+    const arr = cache.apply(
+      `${flag}-${current()}-arr`,
+      _Injector.get(flag).filter(i => i.to === current()).map(i => i.value)
+    );
 
     if (!arr.length) return data;
 
     return data.replace(pattern, str => {
-      const code = arr.reduce((a, c) => a + c, '');
+      const code = cache.apply(
+        `${flag}-${current()}-code`,
+        arr.reduce((a, c) => a + c, '')
```

Comment:
Replace `arr.reduce((a, c) => a + c, '')` to `() => arr.reduce((a, c) => a + c, '')`, If it's not a function, it's immediately executed

Reply:
Rebased and updated.